test/integration/kubernetes_deploy_test.rb

```diff
@@ -1,27 +1,118 @@
 require 'test_helper'
 
 class KubernetesDeployTest < KubernetesDeploy::IntegrationTest
-  def test_basic
-    runner = KubernetesDeploy::Runner.new(
-      namespace: @namespace,
-      current_sha: "abcabcabc",
-      context: MINIKUBE_CONTEXT,
-      template_dir: File.expand_path("./test/fixtures/basic"),
-      wait_for_completion: true,
-    )
-    runner.run
-
-    pods = kubeclient.get_pods(namespace: @namespace)
-    managed_pods, unmanaged_pods = pods.partition do |pod|
-      pod.metadata.ownerReferences && pod.metadata.ownerReferences.first.kind == "ReplicaSet"
+  def test_full_basic_set_deploy_succeeds
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+  end
+
+  def test_partial_deploy_followed_by_full_deploy
+    deploy_fixture_set("basic", ["configmap-data", "redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+    basic.assert_configmap_data_up
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+
+    deploy_fixture_set("basic")
+    basic.assert_all_up
+  end
+
+  def test_pruning
+    deploy_fixture_set("basic")
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_up
+
+    deploy_fixture_set("basic", ["redis"])
+    basic.assert_all_redis_resources_up
+    basic.refute_configmap_data_exists
+    basic.refute_managed_pod_exists
+    basic.refute_web_resources_exist
+  end
+
+  def test_pvcs_are_not_pruned
+    deploy_fixture_set("basic", ["redis"])
+    basic = FixtureSetAssertions::Basic.new(@namespace)
+    basic.assert_all_redis_resources_up
+
+    deploy_fixture_set("basic", ["configmap-data"])
+    basic.assert_configmap_data_up
+    basic.refute_redis_resources_exist(expect_pvc: true)
+  end
+
+  def test_success_with_unrecognized_resource_type
```

Comment:
Secrets _are_ unrecognized. If you run this test with the main logger turned on, you'll see this:

![image](https://cloud.githubusercontent.com/assets/4789493/23238760/8d78ad52-f931-11e6-848e-d590c4d09d34.png)

I can add a comment I guess? I chose them because they're the one thing I don't think we should ever add explicit support for, since it means you're committing secrets to your repo.